event-sourcing/event-source-api/src/test/java/uk/gov/justice/services/eventsourcing/source/api/feed/common/EventsPageServiceTest.java

```diff
@@ -0,0 +1,284 @@
+package uk.gov.justice.services.eventsourcing.source.api.feed.common;
+
+import static java.net.URI.create;
+import static java.util.UUID.randomUUID;
+import static javax.json.Json.createObjectBuilder;
+import static org.hamcrest.CoreMatchers.notNullValue;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.when;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.BACKWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Direction.FORWARD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition.FIRST;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.FixedPosition.HEAD;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.first;
+import static uk.gov.justice.services.eventsourcing.repository.jdbc.Position.head;
+
+import uk.gov.justice.services.common.util.UtcClock;
+import uk.gov.justice.services.eventsourcing.source.api.feed.event.EventEntry;
+import uk.gov.justice.services.eventsourcing.source.api.service.EventsPageService;
+import uk.gov.justice.services.eventsourcing.source.api.service.EventsService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import javax.json.JsonObject;
+
+import org.jboss.resteasy.spi.ResteasyUriInfo;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EventsPageServiceTest {
+
+    @Mock
+    private EventsService service;
+
+    @InjectMocks
+    EventsPageService eventsPageService;
+
+    @Test
+    public void shouldReturnEventsWhenLessRecordsThanPageSizeWhenLookingForNewerEvents() throws Exception {
+
+        final UUID streamId = randomUUID();
+
+        final ResteasyUriInfo uriInfo = new ResteasyUriInfo(create("http://server:123/context/"), create("event-streams/" + streamId));
+
+        final List<EventEntry> events = new ArrayList<>();
+
+        final EventEntry event1 = new EventEntry(randomUUID(), streamId, 3L, "Test Name3", createObjectBuilder().add("field3", "value3").build(), new UtcClock().now().toString());
+
+        events.add(event1);
+
+        when(service.recordExists(4L, streamId)).thenReturn(true);
```

Comment:
spacing makes this harder to read, maybe only have line spaces when you switch between logical blocks.

so group some of the asserts/when's together etc.